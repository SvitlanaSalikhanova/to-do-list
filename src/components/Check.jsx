/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import CheckCss from '../styles/Check.module.scss';

export default function Check({ item, onChange }) {
    return (
        <div className={CheckCss.container}>
            <div className={CheckCss.round}>
                <input
                    type="checkbox"
                    checked={item.isDone}
                    onChange={onChange}
                    id={`checkbox${item.id}`}
                />
                <label htmlFor={`checkbox${item.id}`} />
            </div>
        </div>
    );
}
