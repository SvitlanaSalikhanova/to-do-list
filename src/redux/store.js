import { configureStore } from '@reduxjs/toolkit';
import tasksReducer from './TasksSlice';
import filterReducer from './FilterSlice';

const store = configureStore({
    reducer: {
        tasks: tasksReducer,
        filter: filterReducer,
    },
});

export default store;
