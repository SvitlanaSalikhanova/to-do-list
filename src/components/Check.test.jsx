import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Check from './Check';

describe('Check Snapshot testing suite', () => {
    it('Matches Snapshot', () => {
        const item = {
            id: 1,
            text: 'text',
            isDone: false,
        };
        const domTree = renderer.create(
            <Provider store={store}><Check item={item} /></Provider>,
        ).toJSON();
        expect(domTree).toMatchSnapshot();
    });
});
