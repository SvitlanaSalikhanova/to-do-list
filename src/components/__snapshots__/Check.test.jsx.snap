// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Snapshot testing suite Matches Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="round"
  >
    <input
      checked={false}
      id="checkbox1"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="checkbox1"
    />
  </div>
</div>
`;
